Tests/External/Plugins/ASCompletion.Tests/Completion/ASGeneratorTests.cs

```diff
@@ -0,0 +1,611 @@
+                        }
+                    },
+                    new MemberModel("x", "Int", FlagType.Getter, Visibility.Public)
+                    {
+                        Parameters = new List<MemberModel>
+                        {
+                            new MemberModel {Name = "get"},
+                            new MemberModel {Name = "set"}
+                        }
+                    },
+                    new MemberModel("y", "Int", FlagType.Getter, Visibility.Public)
+                    {
+                        Parameters = new List<MemberModel>
+                        {
+                            new MemberModel {Name = "get"},
+                            new MemberModel {Name = "never"}
+                        }
+                    },
+                    new MemberModel("testMethod", "Float", FlagType.Function, Visibility.Public),
+                    new MemberModel("testMethodArgs", "Int", FlagType.Function, Visibility.Public)
+                    {
+                        Parameters = new List<MemberModel>
+                        {
+                            new MemberModel("arg", "Float", FlagType.Variable, Visibility.Default),
+                            new MemberModel("arg2", "Bool", FlagType.Variable, Visibility.Default)
+                        }
+                    },
+                    new MemberModel("testPrivateMethod", "Float", FlagType.Function, Visibility.Private)
+                    {
+                        Parameters = new List<MemberModel>
+                        {
+                            new MemberModel("?arg", "String", FlagType.Variable, Visibility.Default),
+                            new MemberModel("?arg2", "Int", FlagType.Variable, Visibility.Default)
+                            {
+                                Value = "1"
+                            }
+                        }
+                    }
+                });
+
+
+                return interfaceModel;
+            }
+
+            private ClassModel GetFieldFromParameterClassModel()
+            {
+                var classModel = new ClassModel
+                {
+                    LineFrom = 1,
+                    LineTo = 3
+                };
+
+                classModel.Members.Add(new MemberList
+                {
+                    new MemberModel("FieldFromParameterTest", null, FlagType.Constructor, Visibility.Public)
+                    {
+                        LineFrom = 2, LineTo = 2,
+                        Parameters = new List<MemberModel>
+                        {
+                            new MemberModel {Name = "arg", LineFrom = 2, LineTo = 2}
+                        }
+                    }
+                });
+
+                return classModel;
+            }
+
+            [Test]
+            public void FieldFromParameterPublicScopeWithEmptyBody()
+            {
+                var pluginMain = Substitute.For<PluginMain>();
```

Comment:
I really think the generators tests should be made completely declaratively: input, operation, output.